backend/btrixcloud/webhooks.py

```diff
@@ -0,0 +1,313 @@
+"""Webhook management"""
+
+import asyncio
+from datetime import datetime
+from typing import List
+import uuid
+
+import aiohttp
+import backoff
+from fastapi import APIRouter, Depends, HTTPException
+from pydantic import UUID4
+
+from .pagination import DEFAULT_PAGE_SIZE, paginated_format
+from .models import (
+    WebhookEventType,
+    WebhookNotification,
+    ArchivedItemCreatedBody,
+    CollectionItemAddedRemovedBody,
+    PaginatedResponse,
+    Organization,
+)
+
+
+# ============================================================================
+class EventWebhookOps:
+    """Event webhook notification management"""
+
+    def __init__(self, mdb, org_ops):
+        self.webhooks = mdb["webhooks"]
+        self.colls = mdb["collections"]
+        self.crawls = mdb["crawls"]
+
+        self.org_ops = org_ops
+        self.crawl_ops = None
+
+        self.origin = None
+
+        self.router = APIRouter(
+            prefix="/webhooks",
+            tags=["webhooks"],
+            responses={404: {"description": "Not found"}},
+        )
+
+    def set_crawl_ops(self, ops):
+        """set crawl ops"""
+        self.crawl_ops = ops
+
+    async def list_notifications(
+        self,
+        org: Organization,
+        page_size: int = DEFAULT_PAGE_SIZE,
+        page: int = 1,
+    ):
+        """List all webhook notifications"""
+        # Zero-index page for query
+        page = page - 1
+        skip = page_size * page
+
+        query = {"oid": org.id}
+
+        total = await self.webhooks.count_documents(query)
+
+        cursor = self.webhooks.find(query, skip=skip, limit=page_size)
+        results = await cursor.to_list(length=page_size)
+        notifications = [WebhookNotification.from_dict(res) for res in results]
+
+        return notifications, total
+
+    async def get_notification(self, org: Organization, notificationid: uuid.UUID):
+        """Get webhook notification by id and org"""
+        query = {"_id": notificationid, "oid": org.id}
+
+        res = await self.webhooks.find_one(query)
+        if not res:
+            raise HTTPException(status_code=404, detail="notification_not_found")
+
+        return WebhookNotification.from_dict(res)
+
+    @backoff.on_exception(backoff.expo, aiohttp.ClientError, max_tries=5, max_time=60)
+    async def send_notification(
+        self, org: Organization, notification: WebhookNotification
+    ):
+        """Send notification"""
+        if not org.webhookUrls:
+            print(
+                "Webhook URLs not configured - skipping sending notification",
+                flush=True,
+            )
+            return
+
+        # Get configured URL for this notification type
+        if notification.event == WebhookEventType.ARCHIVED_ITEM_CREATED:
```

Comment:
if the event types are also the keys, could then do something like:
```
if not getattr(org.webhookUrls, notification.event):
   ...
```
and not need to add a conditional for each type of webhook?